core/domain/classifier_domain.py

```diff
@@ -387,51 +355,52 @@ def validate(self):
                     'Expected answers to be a list, received %s' %
                     grouped_answers['answers'])
 
-        # Classifier data can be either None (before its stored) or a dict.
-        if not isinstance(self.classifier_data, dict) and self.classifier_data:
-            raise utils.ValidationError(
-                'Expected classifier_data to be a dict|None, received %s' % (
-                    self.classifier_data))
-
-        if not isinstance(self.data_schema_version, int):
+        if not isinstance(self.algorithm_version, int):
             raise utils.ValidationError(
-                'Expected data_schema_version to be an int, received %s' %
-                self.data_schema_version)
+                'Expected algorithm_version to be an int, received %s' %
+                self.algorithm_version)
 
 
-class TrainingJobExplorationMapping(python_utils.OBJECT):
-    """Domain object for a job-exploration mapping model.
+class StateTrainingJobsMapping(python_utils.OBJECT):
+    """Domain object for a state to training job mapping model.
 
-    A job-exploration mapping is a one-to-one relation between the
-    attributes in an exploration to the training job model for the classifier it
-    needs to use. The mapping is from <exp_id, exp_version, state_name> to the
-    job_id.
+    This is a one-to-many relation between the particular state of the
+    particular version of the particular exploration and classifier training
+    jobs. Each <exp_id, exp_version, state_name> is mapped to a
+    algorithm_id_to_job_id dict which maps all the valid algorithm_ids for the
+    given state to their training jobs. A state may have multiple algorithm_ids
+    valid for it for example, one algorithm would serve Oppia web users while
+    other would support Oppia mobile or android user. How many algorithm_ids are
```

Comment:
... while another might support Oppia mobile or Android users. The number of algorithm_ids that are ...

Reply:
Done